### DISCUSSION TITLE
Is it possible to track atoms and their interaction in docking poses after return_atom is removed.

### DISCUSSION THREAD

**Discussion by syedzayyan** [chemosim-lab/ProLIF#166] [General]
Hello!

In the previous version, there was used to be an option for return_atom when generating data frames. Basically, I am interested in one atom in the whole molecule and I want to check its interaction over several docking poses. Is that really feasible?

**Comment by cbouy** [comment]
Hi @syedzayyan,

If you really want to work on a dataframe you can use something like in [this comment](https://github.com/chemosim-lab/ProLIF/issues/164#issuecomment-1767279203).

Otherwise this metadata is in the `fp.ifp` object which you can traverse like so (if I understand what you're looking for correctly):

```python
from collections import defaultdict

atom_mapping = defaultdict(list)

for pose_index, pose_ifp in fp.ifp.items():
    for (_, resid), residue_ifp in pose_ifp.items():
        for interaction_name, metadata_tuple in residue_ifp.items():
            ligand_atoms = set([metadata["parent_indices"]["ligand"] for metadata in metadata_tuple])
            for atom in ligand_atoms:
                atom_mapping[atom].append({"residue": str(resid), "interaction": interaction_name, "pose_index": pose_index})
```

Please close the issue if this answers your question

**Reply by syedzayyan** [reply]
Thanks @cbouy for the answer. I know I am dragging this a bit but do you already have the Euclidean distance between the ligand atom and the atom it interacts with on the ligand? In the `count=True` section of the docs, there is a mention that one can check for distances to select poses.

**Reply by syedzayyan** [reply]
Nevermind, I have found it. For anyone wandering everything including the angles are in the interaction data!
